helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java

```diff
@@ -459,14 +520,21 @@ public void updateDeploymentGroupHosts(final String groupName, final List<String
         final List<ZooKeeperOperation> ops = Lists.newArrayList();
         ops.add(set(Paths.statusDeploymentGroupHosts(groupName), Json.asBytes(hosts)));
 
-        final DeploymentGroup deploymentGroup = getDeploymentGroup(groupName);
+        DeploymentGroup deploymentGroup = getDeploymentGroup(groupName);
         ImmutableList<Map<String, Object>> events = ImmutableList.of();
 
         if (deploymentGroup.getJobId() != null) {
-          final DeploymentGroupStatus deploymentGroupStatus = getDeploymentGroupStatus(groupName);
-          if (deploymentGroupStatus == null || deploymentGroupStatus.getState() != FAILED) {
-            final RollingUpdateOp op =
-                getInitRollingUpdateOps(deploymentGroup, hosts, HOSTS_CHANGED, client);
+          if (updateOnHostChange(deploymentGroup, status)) {
```

Comment:
I think there's still a race condition here, e.g.:
 * Get DG and DG status [in the context of this code]
 * Manual rolling-update triggered [concurrently with this code running]
 * `updateOnHostChange()` is run, and returns true
 * --> Current [manual] rolling-update is overwritten

You could add a ZK `check()` in the transaction to ensure that the current version of the DG config and/or status node has not changed if you want to avoid the race.